api/v3/Membership.php

```diff
@@ -33,6 +33,23 @@
  */
 
 /**
+ * Adjust Metadata for Delete action.
+ *
+ * The metadata is used for setting defaults, documentation & validation.
+ *
+ * @param array $params
+ *   Array of parameters determined by getfields.
+ */
+function _civicrm_api3_membership_delete_spec(&$params) {
+  $params['preserve_contribution'] = array(
+    'api.required' => 0,
+    'title' => 'Preserve Contribution',
+    'description' => 'By default this is 0, or 0 if not set. Set to 1 to preserve the associated contribution record when membership is deleted.',
```

Comment:
My concern with @eileenmcnaughton suggestion is that it changes the current behavior. If someone has existing code that deletes a membership with an associated contribution, they will be expecting that contribution to also be deleted. I can see preserving the contribution being a desired behavior, but only if this were a new API or BAO action. My goal was to add this improvement without changing the way any existing calls this this API or BAO behave.